### PR TITLE
Output pdf dicts in deterministic order

### PR DESCRIPTION
**Pull request by jkseppan**
For #6317 (but not a complete solution for that)

### CODE REVIEW

**Comment by tacaswell** [comment]
Should we merge this as is or do you intend to add more commits to this?

File: lib/matplotlib/backends/backend_pdf.py [hunk 1]
```diff
@@ -182,8 +183,8 @@ def pdfRepr(obj):
     # represented as Name objects.
     elif isinstance(obj, dict):
         r = [b"<<"]
-        r.extend([Name(key).pdfRepr() + b" " + pdfRepr(val)
-                  for key, val in six.iteritems(obj)])
+        r.extend([Name(key).pdfRepr() + b" " + pdfRepr(obj[key])
+                  for key in sorted(six.iterkeys(obj))])
```
Comment (QuLogic) [inline lib/matplotlib/backends/backend_pdf.py]: The `iterkeys` seems unnecessary if you're going to sort it anyway; `sorted(obj)` should work just fine.

Comment (jkseppan) [inline lib/matplotlib/backends/backend_pdf.py]: Thanks, that makes sense.

**Comment by jkseppan** [comment]
@tacaswell I think it's probably better to merge sooner, it's going to be some work to implement the rest and this could make it easier for someone else to get started.

**Pull request merged**